engine/src/main/java/org/terasology/world/block/BlockArea.java

```diff
@@ -0,0 +1,622 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.block;
+
+import com.google.common.base.Preconditions;
+import org.joml.AABBf;
+import org.joml.AABBi;
+import org.joml.Math;
+import org.joml.Rectangled;
+import org.joml.Rectanglef;
+import org.joml.Rectanglei;
+import org.joml.RoundingMode;
+import org.joml.Vector2fc;
+import org.joml.Vector2i;
+import org.joml.Vector2ic;
+import org.joml.Vector3ic;
+
+import java.util.Objects;
+
+/**
+ * A bounded axis-aligned rectangle of blocks.
+ * <p>
+ * A block area is described and backed by an {@link org.joml.Rectanglei}
+ *
+ * @see BlockAreas
+ */
+public class BlockArea {
+    private final Vector2i min = new Vector2i(Integer.MAX_VALUE);
+    private final Vector2i max = new Vector2i(Integer.MIN_VALUE);
+
+    private boolean dirty = true;
+    private Rectanglef rectangle = new Rectanglef();
```

Comment:
@pollend Should we compute the bounding rectangle every time something changes, or just when it is requested (lazy)? I've prepared to do the latter, not fully implemented yet, though.

Reply:
I would just update the bounds if it changes. you still end up needing to recalculate it for most cases.

Reply:
I would expect this to be used sufficiently rarely that just not cacheing at all would be best.

Reply:
If we're going to re-compute it anyways, maybe make it more explicit? Then the caller can do whatever they want with it....

```java
Rectanglef worldArea = BlockAreas.toWorldArea(area);
```